java-frontend/src/main/java/org/sonar/java/se/CheckerDispatcher.java

```diff
@@ -94,6 +98,9 @@ public void addTransition(ProgramState state) {
 
   private void executePost() {
     this.transition = false;
+    if (!syntaxNode.is(Tree.Kind.METHOD_INVOCATION)) {
```

Comment:
as discussed, due to current implementation, identifier tree of the method will be evaluated before the actual invocation, which will assure resetting of the chosen field, so the above mentioned case can not actually happen. However this is coincidence and should be not relied upon. I moved resetting the field to more appropriate place and I added the testcase verifying the behavior